### ISSUE TITLE
Invalid telemetry data for Arc mobile

### ISSUE THREAD

**Issue by DevTechProfile** [intel/drivers.gpu.control-library#33]
Our telemetry implementation works fine for Arc desktop cards like A770. 

Code: https://github.com/CXWorld/CapFrameX/blob/release/1.7.1/source/CapFrameX.IGCL/IGCLManager.cpp

But for a mobile GPU like A770M the sensors calculated via (current - prev)/deltatime (TDP, loads) are invalid.

![image](https://user-images.githubusercontent.com/30285858/206264268-6a6641bd-ff39-45a9-bc4e-b53619ca684c.png)

**Comment by ezhilintc** [comment]
Please make sure, the adapter queried is accurate, as there might be multiple Intel adapters (integrated and discrete) in mobile platforms. Could you please specify, which API's are reporting incorrect Values?

**Comment by DevTechProfile** [comment]
We have a filter for iGPUs so we are talking about sensors of the A770M. For example, the GPU TDP sensor is wrong. 

Your code:
`
ctl_result_t status = ctlPowerTelemetryGet(hDevices[index], &pPowerTelemetry);

if (status == ctl_result_t::CTL_RESULT_SUCCESS)
{
	prevtimestamp = curtimestamp;
	curtimestamp = pPowerTelemetry.timeStamp.value.datadouble;
	deltatimestamp = curtimestamp - prevtimestamp;

	if (pPowerTelemetry.gpuEnergyCounter.bSupported)
	{
		telemetryData->gpuEnergySupported = true;
		prevgpuEnergyCounter = curgpuEnergyCounter;
		curgpuEnergyCounter = pPowerTelemetry.gpuEnergyCounter.value.datadouble;

		telemetryData->gpuEnergyValue = (curgpuEnergyCounter - prevgpuEnergyCounter) / deltatimestamp;
	}
}
`
It works fine with the A770 desktop card.

**Comment by DevTechProfile** [comment]
@pmmeyer 

1. All sensors calculated by (current - prev)/deltatime (TDP, loads) are invalid.
2. 1s refresh rate
3. Idle and under load

Thanks for looking at this.

**Comment by pmmeyer** [comment]
@DevTechProfile  - can you please confirm which graphics driver you are using?

**Comment by DevTechProfile** [comment]
Yes, on the left side you find the used driver version.

![image](https://user-images.githubusercontent.com/30285858/218754704-f6838c93-5e36-402e-bb3d-9930dc7b8b0c.png)

**Comment by pmmeyer** [comment]
This appears to be an issue with CapFrameX's handling of the two display adapters.

Using the same `ctlPowerTelemetryGet` API, I wrote a sample application to log power and render activity for both the discrete and integrated display adapters. I ran my app and also used CapFrameX's capture tool to log the same metrics while running 3DMark TimeSpy.

Here is the output from my app:
![image](https://user-images.githubusercontent.com/73550131/225757476-fb9312c8-9a57-48e3-8fe1-f3147953d67c.png)

CapFrameX's data are invalid though - negative numbers and very large numbers that are junk.

One thing I noticed is that for a given metric (e.g. power) the iGPU and dGPU numbers are almost identical.

Another thing I noticed was that if I disabled the integrated adapter in device manager, launched CapFrameX so that it only detected the discrete adapter, and then re-enabled the integrated adapter, I was able to run TimeSpy and get good data reported by CapFrameX for the discrete adapter. But whenever launching CapFrameX with both adapters enabled, the data get corrupted. 

I would start debugging by logging all of the telemetry returned by the API and manually computing power/activity to validate that the data by the API is valid and then dig deeper to see if that data is somehow being copied/accessed by the wrong adapter when processing.
